felix/bpf/ut/map_upgrade_test.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2022 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ut_test
+
+import (
+	"os"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/projectcalico/calico/felix/bpf"
+	mock "github.com/projectcalico/calico/felix/bpf/mock/multiversion"
+	v2 "github.com/projectcalico/calico/felix/bpf/mock/multiversion/v2"
+	v3 "github.com/projectcalico/calico/felix/bpf/mock/multiversion/v3"
+	v4 "github.com/projectcalico/calico/felix/bpf/mock/multiversion/v4"
+	v5 "github.com/projectcalico/calico/felix/bpf/mock/multiversion/v5"
+)
+
+const key = 0xdeadbeef
+const val = 0xa0b1c2d3
+
+func deleteMap(bpfMap bpf.Map) {
+	bpfMap.(*bpf.PinnedMap).Close()
+	os.Remove(bpfMap.Path())
+}
+
+func TestMapUpgradeV2ToV3(t *testing.T) {
```

Comment:
I am not sure if we are using the caching map because the new map is already online. If so, shouldn't we also test an external update to the map while the the upgrade is in progress?

Reply:
yes, the CT specific stuff will be done in a separate PR